realm/src/main/java/io/realm/Realm.java

```diff
@@ -656,6 +687,15 @@ private static void validateAgainstExistingConfigurations(RealmConfiguration new
                 throw new IllegalArgumentException("Two configurations with different schemas are trying to open " +
                         "the same Realm file. Their schema must be the same: " + newConfiguration.getPath());
             }
+
+            // Check if the durability is the same
+            SharedGroup.Durability cachedDurability = cachedConfiguration.getDurability();
+            SharedGroup.Durability newDurability = newConfiguration.getDurability();
+            if (cachedDurability.value != newDurability.value) {
+                throw new IllegalArgumentException("Configurations cannot have different durabilities for the " +
+                        "same realm filename/identifier: " + newConfiguration.getRealmFileName() + " " +
```

Comment:
We were worried that talking about the path of an in-memory realm would confuse users, but we've ended up getting all of that confusion anyway. We've talked about switching from "in memory" to "transient" or something along those lines.